lib/charms/vault_k8s/v0/vault_kv.py

```diff
@@ -0,0 +1,682 @@
+            return
+        if not relation:
+            logger.warning("Relation is None")
+            return
+        if not relation.active:
+            logger.warning("Relation is not active")
+            return
+        relation.data[self.charm.app]["ca_certificate"] = ca_certificate
+
+    def set_mount(self, relation: ops.Relation, mount: str):
+        """Set the mount on the relation."""
+        if not self.charm.unit.is_leader():
+            return
+
+        relation.data[self.charm.app]["mount"] = mount
+
+    def set_egress_subnets(self, relation: ops.Relation, egress_subnets: List[str]):
+        """Set the egress_subnets on the relation."""
+        if not self.charm.unit.is_leader():
+            return
+        relation.data[self.charm.app]["egress_subnet"] = ",".join(egress_subnets)
+
+    def set_unit_credentials(
+        self,
+        relation: ops.Relation,
+        nonce: str,
+        secret: ops.Secret,
+    ):
+        """Set the unit credentials on the relation."""
+        if not self.charm.unit.is_leader():
+            return
+
+        credentials = self.get_credentials(relation)
+        if secret.id is None:
+            logger.debug(
+                "Secret id is None, not updating the relation '%s:%d' for nonce %r",
+                relation.name,
+                relation.id,
+                nonce,
+            )
+            return
+        credentials[nonce] = secret.id
+        relation.data[self.charm.app]["credentials"] = json.dumps(credentials, sort_keys=True)
+
+    def remove_unit_credentials(self, relation: ops.Relation, nonce: str | Iterable[str]):
+        """Remove nonce(s) from the relation."""
+        if not self.charm.unit.is_leader():
+            return
+
+        if isinstance(nonce, str):
+            nonce = [nonce]
+
+        credentials = self.get_credentials(relation)
+
+        for n in nonce:
+            credentials.pop(n, None)
+
+        relation.data[self.charm.app]["credentials"] = json.dumps(credentials, sort_keys=True)
+
+    def get_credentials(self, relation: ops.Relation) -> dict:
+        """Get the unit credentials from the relation."""
+        return json.loads(relation.data[self.charm.app].get("credentials", "{}"))
+
+    def get_outstanding_kv_requests(self, relation_id: int | None = None) -> List[KVRequest]:
+        """Get the outstanding requests for the relation."""
+        outstanding_requests: List[KVRequest] = []
+        kv_requests = self.get_kv_requests(relation_id=relation_id)
+        for request in kv_requests:
+            if not self._credentials_issued_for_request(
+                nonce=request.nonce, relation_id=request.relation_id
+            ):
+                outstanding_requests.append(request)
+        return outstanding_requests
+
+    def get_kv_requests(self, relation_id: int | None = None) -> List[KVRequest]:
+        """Get all KV requests for the relation."""
+        kv_requests: List[KVRequest] = []
+        relations = (
```

Comment:
The same comment and in in the autoseal lib applies here

Reply:
fixed in #499